pkg/scheduler/framework/v1alpha1/framework.go

```diff
@@ -583,7 +583,7 @@ func (f *framework) GetWaitingPod(uid types.UID) WaitingPod {
 func pluginNameToConfig(args []config.PluginConfig) map[string]*runtime.Unknown {
 	pc := make(map[string]*runtime.Unknown, 0)
 	for _, p := range args {
-		pc[p.Name] = &p.Args
+		pc[p.Name] = p.Args.DeepCopy()
```

Comment:
IMO `DeepCopy()` can be eliminated by:

```go
	for i := range args {
		p := args[i]
		pc[p.Name] = &p.Args
	}
```

or

```go
	for _, p := range args {
		p := p
		pc[p.Name] = &p.Args
	}
```

Reply:
strange, I thought a left a comment to that affect, I probably forgot to click the submit button!

Reply:
I prefer the first approach

```go
	for i := range args {
		p := args[i]
		pc[p.Name] = &p.Args
	}
```

Reply:
or you could do to save a line :)

```
for p, i := range args {
	pc[p.Name] = &args[i].Args
}
```